lib/remote/configpackageutility.cpp

```diff
@@ -196,9 +197,15 @@ void ConfigPackageUtility::TryActivateStageCallback(const ProcessResult& pr, con
 			<< "Config validation failed for package '"
 			<< packageName << "' and stage '" << stageName << "'.";
 	}
+
+	// Don't let the deferred callback go out of scope when the config validation was successful
+	// and the Icinga2 worker process is going to be reloaded
+	if (reload && pr.ExitStatus == 0) {
```

Comment:
Given that this is now the same condition as where `Application::RequestRestart();` is called, please move it there.

Reply:
Not at all! The `activate` param is also required to be `true`, so that the application restart can be triggered.

Reply:
I'd expect that further requests are only prevented if the node actually reloads immediately. Shouldn't this be the case if and only if `Application::RequestRestart();` is called above?